docs/development/workflow/development_workflow.rst

```diff
@@ -535,27 +535,31 @@ for further information.
 
 .. _rebase:
 
-Rebase, but only if asked
-*************************
+Rebase if necessary
+*******************
+
+Rebasing means taking your changes and applying them to the latest
+version of the ``main`` branch of the official ``astropy`` repository as though that was the
+version you had originally branched from. Each individual commit remains
+visible, but with new commit hashes.
 
-Sometimes the maintainers of Astropy may ask a pull request to be *rebased*
-or *squashed* in the process of reviewing a pull request for merging into
-the main Astropy *main* repository.
+When to rebase
+==============
 
-The decisions of when to request a *squash* or *rebase* are left to
-individual maintainers.  These may be requested to reduce the number of
-visible commits saved in the repository history, or because of code changes
-in Astropy in the meantime.  A rebase may be necessary to allow the Continuous
-Integration tests to run.  Both involve rewriting the `git`_ history, meaning
-that commit hashes will change, which is why you should do it only if asked.
+Pull requests **must** be rebased (but not necessarily squashed to a single
+commit) if at least one of the following conditions is met:
 
-Conceptually, rebasing means taking your changes and applying them to the latest
-version of the development branch of the official Astropy as though that was the
-version you had originally branched from. Each individual commit remains
-visible, but with new metadata/commit hashes. Squashing commits changes the
-metadata/commit hash, and also removes separate visibility of individual
-commits; a new commit and commit message will only contain a textual
-list of the earlier commits.
+* There are conflicts with ``main``.
+* There are commits in ``main`` (after the PR branch point) needed for continuous
+  integration tests to run correctly.
+* There are merge commits from ``astropy/main`` in the PR commit history (merge
+  commits from PRs to the user's fork are fine).
+* There are commit messages that violate the code of conduct.
```

Comment:
Rebasing won't get rid of offensive commit messages, so not sure why this was added.

```suggestion
```

Reply:
Unless you also mean interactive rebase to squash?